addons/website/static/src/components/switch/switch.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module **/
+
+const { Component, tags } = owl;
+
+export class Switch extends Component {
+    setup() {
+        this.extraClasses = this.props.extraClasses ? ` ${this.props.extraClasses}` : '';
+    }
+}
+Switch.props = {
+    value: Boolean,
+    extraClasses: String,
+    label: {type: String, optional: true},
+};
+Switch.template = tags.xml`
+<label t-att-class="'o_switch' + extraClasses">
+    <input type="checkbox" t-att-checked="props.value"/>
+    <span/>
+    <div t-if="props.label"><t t-esc="props.label"/></div>
+</label>
+`;
```

Comment:
Are you sure the backend does not already have its own switches? I had that in mind

Reply:
I found a checkbox, but not a switch as we want it in the website, that could be used for the publish button as well